### PR TITLE
Make generic version, version finder, version selection policy

### PR DESCRIPTION
**Pull request by ydai1124**
@ibuenros  Can you review? Thanks

### CODE REVIEW

File: gobblin-data-management/src/main/java/gobblin/data/management/version/finder/VersionFinder.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import gobblin.dataset.FileSystemDataset;
+import gobblin.data.management.version.FileSystemDatasetVersion;
+
+
+/**
+ * Finds dataset versions.
+ *
+ * @param <T> Type of {@link DatasetVersion} expected from this class.
+ */
+public interface VersionFinder<T extends FileSystemDatasetVersion> {
```
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/version/finder/VersionFinder.java]: Does this need to be restricted to `FileSystemDatasetVersion`?

**Comment by ibuenros** [comment]
LGTM other than one comment

**Comment by ydai1124** [comment]
@ibuenros addressed comment.

**Comment by ibuenros** [comment]
LGTM

**Pull request merged**